cryptography/src/ed25519/mod.rs

```diff
@@ -1,4 +1,4 @@
-//! Ed25519 implementation of the [crate::Signer] trait.
+//! Ed25519 implementation of the [crate::SigningScheme] trait.
```

Comment:
Should we add an impl `Scheme` in these files to enforce we've defined everything?

Reply:
For example, we do the following in `Digest` for `Codec` "joint" traits:

https://github.com/commonwarexyz/monorepo/blob/4e9f32114c51ebc3f7aac0c24e7064c5b295af45/cryptography/src/sha256/mod.rs#L123-L125

Reply:
Those types also benefit from the blanket impl. We just include it as a marker/compile-time check to make sure we don't fuck that up.

Reply:
Example of a blanket: https://github.com/commonwarexyz/monorepo/blob/4e9f32114c51ebc3f7aac0c24e7064c5b295af45/codec/src/codec.rs#L85-L86

Reply:
Although, I suppose we don't really assert types implement `Codec` using this trick. Seems to just be something we do for `Array`. 🤔